validate/error.go

```diff
@@ -0,0 +1,69 @@
+package validate
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ComplianceLevel represents the OCI compliance levels
+type ComplianceLevel int
+
+const (
+	ComplianceOptional ComplianceLevel = iota
+	ComplianceMay
+	ComplianceRecommended
+	ComplianceShould
+	ComplianceShouldNot
+	ComplianceShall
+	ComplianceShallNot
+	ComplianceRequired
+	ComplianceMustNot
+	ComplianceMust
+)
+
+// OCIErrorCode represents the compliance content
+type OCIErrorCode int
+
+const (
+	DefaultFilesystems OCIErrorCode = iota
+)
+
+// OCIError represents an error with compliance level and OCI reference
+type OCIError struct {
+	Level     ComplianceLevel
+	Reference string
+	Err       error
+}
+
+//FIXME: change to tagged spec releases
```

Comment:
> So each OCI Error will be associate with one or more spec versions.

I think it's better to only test against the spec for the target version.  But if we're passing around that target version so we can decide what conditions to test, there's probably no need to stash it on `Error`.  I'll withdraw my suggestion, and we can leave the struct fields alone.